crates/ide-diagnostics/src/handlers/type_mismatch.rs

```diff
@@ -184,6 +189,86 @@ fn add_missing_ok_or_some(
     Some(())
 }
 
+fn remove_unnecessary_wrapper(
+    ctx: &DiagnosticsContext<'_>,
+    d: &hir::TypeMismatch,
+    expr_ptr: &InFile<AstPtr<ast::Expr>>,
+    acc: &mut Vec<Assist>,
+) -> Option<()> {
+    let root = ctx.sema.db.parse_or_expand(expr_ptr.file_id);
+    let expr = expr_ptr.value.to_node(&root);
```

Comment:
This node might be inside a macro expansion which will cause issues later as we aren't accounting for that in our replacements so we should attempt to upmap the node via `ctx.sema.original_ast_node(&expr)?`